### DISCUSSION TITLE
How do I find the range of my variogram?

### DISCUSSION THREAD

**Discussion by nkruskamp** [GeoStat-Framework/GSTools#142] [Q&A] [answered]
Hi,

A few questions. Please let me know if I need to submit different discussions for each.

1. When I fit a variogram, I am able to access nugget and sill properties of the variogram object. Is there a similar property for the range? Is it somehow related to the len_scale?
2. My assumption is that the bins represent euclidean distance in the unit of my original data when estimating the variogram from unstructured data. Is there a way to specify bins and a distance between bins?

Thanks!

**Comment by LSchueler** [comment, marked as answer]
Hi!
I think we'll be fine discussing both issues here.

1. The `len_scale` is the same as the range. The naming conventions differ the most for this parameter across the different communities and we think that len_scale is a bit more descriptive than range, but that's a personal choice I guess ;-)
2. Exactly, normally the bins give the Euclidean distance in the units of your choice (or rather your data). You can use spherical coordinates, but I understand that that is not your main concern. If you want to give explicit bins, simply hand them over to the estimation method, the keyword is `bin_edges`. You can have a look at this [tutorial](https://geostat-framework.readthedocs.io/projects/gstools/en/latest/examples/03_variogram/03_directional_2d.html#sphx-glr-examples-03-variogram-03-directional-2d-py). In the 3rd code snippet box, the bins array is defined in the first line and then handed over to the variogram estimation method in line 2. The example is for spherical coordinates, but it works exactly the same in Euclidean coordinates.

**Reply by nkruskamp** [reply]
Thanks so much for the fast reply!

1. I assumed the len_scale is the value I am looking for. I think my confusion comes from the `len_scale` value not lining up with what I would expect the value to be by looking at the variogram. I thought the `len_scale`/range value should be where the variogram curve asymptotes at the sill? I've attached an image to show what I expected versus what I'm seeing. I've tried a few models and the result appears to be the same.

For reference, I'm using version 1.2.1 in a conda environment, package downloaded via conda-forge yesterday.

![ex_vario](https://user-images.githubusercontent.com/26750564/111358308-933a8380-8660-11eb-8326-a30824243060.png)


2. I see now. I was using a code example from your documentation using `np.arange` but didn't think to include the `step` parameter as a way to specify the bin width. That's exactly what I was looking for.

**Reply by MuellerSeb** [reply]
Hey there! Interpreting the range / len_scale is always tricky, especially when using different models. We wrote a separate tutorial about this issue: [Different scales](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/examples/02_cov_model/04_different_scales.html#sphx-glr-examples-02-cov-model-04-different-scales-py)

Practitioners often use the percentile scale to determine, at which distance 90% (for example) of the sill is reached:
```python
import gstools as gs
model = gs.Stable(len_scale=10)
print("90% Percentile scale:", model.percentile_scale(0.9))
```
```bash
90% Percentile scale: 17.437215135964117
```
As you see, Percentile scale for 90% is almost two times the length scale.

The Integral scale is another scale definition, that is independent of any specific model definition.

Note that this issue is one of the main reasons, why the spherical model is favored by a lot of people, since the length scale is the biggest distance of any positive correlation. And that is totally easy to interpret.

Cheers,
Sebastian

**Reply by nkruskamp** [reply]
Thanks so much for clarifying! I know only the basics of variograms right now so this is very helpful.

re: spherical models. I have seen related studies use spherical models, but for some reason I cannot get them to fit my data. (I get a straight line variogram). As you can see in the image I shared, my data are very noisy and I originally tested it but have left it out thus far since it isn't providing meaningful results. Other models do seem to fit pretty well.  It also appears related to the bin width, so I am trying to figure out how to solve that issue as well.

I may be back with a separate question related to binning and model performance. I see code here related to binning and model scoring so I look forward to the 1.3.0 release when it's ready!
